features/step_definitions/web_steps.rb

```diff
@@ -143,14 +143,10 @@ def with_scope(locator)
 
 Then /^(.+) and confirm the dialog(?: "(.*)")?$/ do |original, text|
   ActiveSupport::Deprecation.warn "🥒 Replace with step 'And confirm the dialog'"
-  if rack_test?
+  accept_confirm(text) do
     step original
-  else
-    accept_confirm(text) do
-      step original
-    end
-    wait_for_requests
   end
+  wait_for_requests
```

Comment:
I need some help to understand this

Reply:
I don't quite understand it either, I just copy-pasted it from @josemigallas 's PR :)

Reply:
maybe internally `accept_confirm` does nothing in rack_test so no need to complicate the code here, just a guess

Reply:
I think the thing here is that `rack_test` doesn't support modals, and it throws an exception when trying to call it: https://github.com/teamcapybara/capybara/blob/3.40.0/lib/capybara/driver/base.rb#L138-L140

But I think that this step was just simplified, because any test where you would expect to have a modal window (and thus where you would use this step) will need to be `@javascript` anyway. Nowadays, I think any admin portal tests MUST have `@javascript`, because even the login form is JS-based, and most of the menus etc.

So I think each time we have less non-JS tests, so this is why we are simplifying.